examples/step-44/step-44.cc

```diff
@@ -2470,6 +2470,17 @@ namespace Step44
     constraints.clear();
     const bool apply_dirichlet_bc = (it_nr == 0);
 
+    // In this particular example, the boundary values will be calculated for
+    // the two first iterations of Newton's algorithm. In general, one would
+    // build non-homogeneous constraints in the first iteration (that is, when
+    // `apply_dirichlet_bc == true`) and build only the homogeneous constraints
+    // in the following step.  While the current example has only homogenous
+    // constraints, previous experiences have shown that an common error is
+    // forgetting to add the extra condition when refactoring the code to
+    // specific uses. This could lead to hard to debug errors. In this spirit,
+    // we chose to make the code easier to debug, even if at the cost of a bit
```

Comment:
"make the code more verbose in terms of what operations are performed at each Newton step"

I think that this is a more precise reason as to why the code is not optimally concise. Do you agree with this?